docs/parameters.md

```diff
@@ -0,0 +1,188 @@
+# Passing parameters to brokers
+
+Table of Contents
+- [Overview](#overview)
+- [Design](#design)
+  - [Basic example](#basic-example)
+  - [Passing sensitive data](#passing-sensitive-data)
+    - [Reference to a secret](#reference-to-a-secret)
+    - [The entire payload with sensitive data](#the-entire-payload-with-sensitive-data)
+  - [Merging multiple sources and conflict resolution](#merging-multiple-sources-and-conflict-resolution)
+- [Example with multiple sources](#example-with-multiple-sources)
+
+## Overview
+`parameters` and `parametersFrom` properties of `Instance` and `Broker` resources 
+provide support for passing parameters to the broker relevant to the corresponding
+[provisioning](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#provisioning) or
+[binding](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#binding) request. 
+The resulting structure represents an arbitrary JSON object, which is assumed to 
+be valid for a particular broker. 
+The Service Catalog does not enforce any extra limitations on the format and content 
+of this structure.
+
+## Design
+
+When you create an `Instance` or a `Binding`, you can set parameters to be passed 
+to the corresponding broker.
+To set parameters, include the `parameters` or `parametersFrom` field in the spec.
+
+### Basic example
+
+Let's say we want to create an `Instance` of EC2 running on AWS using a
+[corresponding broker](https://github.com/cloudfoundry-samples/go_service_broker) 
+which implements the Open Service Broker API.
+
+A typical provisioning request for this broker looks [like this](https://github.com/cloudfoundry-samples/go_service_broker/blob/master/bin/curl_broker.sh):
+```bash
+curl -X PUT http://username:password@localhost:8001/v2/service_instances/instance_guid-111 -d '{
+  "service_id":"service-guid-111",
+  "plan_id":"plan-guid",
+  "organization_guid": "org-guid",
+  "space_guid":"space-guid",
+  "parameters": {"ami_id":"ami-ecb68a84"}
+}' -H "Content-Type: application/json"
+```
+
+Note that the broker accepts an `ami_id` parameter ([AMI](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) 
+identifier).
+To configure a provisioning request in Service Catalog, we need to declare an `Instance` 
+resource with an AMI identifier declared in the `parameters` field of its spec:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: ami-instance
+  namespace: test-ns
+spec:
+  serviceClassName: aws-ami
+  planName: default
+  parameters:
+    ami_id: ami-ecb68a84
+```
+
+### Passing sensitive data
+
+`Secret` resources can be used to store sensitive data. The `parametersFrom`
+field allows to reference the external parameters source.
+The following sections describe the two ways in which secrets can be used to 
+populate the parameters sent to a broker.
+
+#### Reference to a secret
+
+Use the `secretRef` field to pass a reference to a secret with parameter contents.
```

Comment:
(re-adding this comment here since the original one was lost during a previous update to the doc)

Based on what @pmorie indicated was agreed to in the last call ( #1079 (comment) ) it seems like the current plan is to remove support for secret map[string]string for now - along with parameters. That would at least be consistent to not support map[string]string regardless of whether its in a secret or not.

However, based on what I've seen/heard it seems that there might not be agreement on what was decided during the last call. I was going to be on today's call to hear first-hand the conversation about this but it was cancelled. If everything goes as planned I should be on plane during tomorrow's call so I won't be able to make it but I would like to suggest a few things:

1 - based on paul's comment that I referenced above, it seems like (for now) we'll just support the user giving us a blob of json (either in or out of a secret), period. Nothing else. I view this as the absolute minimum we need since I kind of view everything else as UX niceties - so them giving us a blob is the ultimate "escape hatch" when they just can't do what they want in a nicer UX way. Can we get agreement on whether this is (1) what is being agreed to, and (2) the rationale behind the decision aligns with what I said. I think the rationale part is important because it goes to understanding the usecases/requirements we're trying to address with this decision. W/o that I feel like we may be just picking a solution w/o a clear understanding of why or what problem it is trying to solve so we can't gauge whether we've met our goals or not.

2 - if the previous point is correct, then I think its important that we start to gather a list of usecases/requirements that we think follow-on PRs in this space should address. After all of our talks I'm not convinced that everyone has the same list of goals and it would be good to enumerate them and agree on which ones we want to solve vs discard. Hashing thru this list would be an excellent f2f topic. What did we ever decide about another f2f? I just can't recall.